tasks/ops-agent.yml

```diff
@@ -0,0 +1,62 @@
+---
+- name: Validate package_state
+  assert:
+    that: package_state in ['present', 'absent']
+    msg: "Received invalid package state: '{{ package_state }}'. The Cloud Ops Ansible role supports the following package states: 'present' and 'absent'."
+
+- name: Create temp directory
+  tempfile:
+    path: /tmp
+    state: directory
+    suffix: cloud_ops_shell_scripts
+  register: tempfolder
+  check_mode: no
+  changed_when: false
+
+- name: Download script
+  get_url:
+    url: https://dl.google.com/cloudagents/add-google-cloud-ops-agent-repo.sh
+    dest: "{{ tempfolder.path }}/add-google-cloud-ops-agent-repo.sh"
+    mode: 0755
+  check_mode: no
+  changed_when: false
+
+- when: package_state == 'present'
+  block:
+  - name: Add repo and install agent
+    command:
+      chdir: "{{ tempfolder.path }}"
+      cmd: "bash add-google-cloud-ops-agent-repo.sh --also-install --version={{ version }} {{ '--dry-run' if ansible_check_mode else '' }}"
+    environment:
+      REPO_CODENAME: "{{ ansible_distribution_release }}"
+    register: result
+    check_mode: no
+    changed_when: "'No changes made.' not in result.stdout_lines"
+    notify: restart google-cloud-ops agent
+
+  - name: Copy main config file onto the remote machine
+    copy:
+      src: "{{ main_config_file }}"
+      dest: /etc/google-cloud-ops-agent/config.yaml
+      force: yes
+      mode: 0644
+      validate: '/opt/google-cloud-ops-agent/subagents/collectd/sbin/collectd -tC %s; /opt/google-cloud-ops-agent/subagents/fluent-bit/bin/fluent-bit -c %s'
+    when: main_config_file|length > 0
+    notify: restart google-cloud-ops agent
+
```

Comment:
Does the ops agent support additional configs? If so, in which directory are they stored?

Reply:
Done.